valhalla/mjolnir/landmark_database_builder.h

```diff
@@ -15,55 +16,16 @@ struct Landmark {
 
 struct LandmarkDatabase {
 public:
-  LandmarkDatabase(const std::string& db_name, bool read_only) : db(nullptr), database(db_name) {
-    if (!filesystem::exists(database)) {
-      if (read_only) {
-        throw std::runtime_error("invalid option");
-      }
-      open_flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
-      connect_database();
-    } else {
-      open_flags = read_only ? SQLITE_OPEN_READONLY : SQLITE_OPEN_READWRITE;
-      connect_database();
-    }
-  }
-
-  ~LandmarkDatabase() {
-    if (did_inserts && !vacuum_analyze()) {
-      LOG_ERROR("cannot do vacuum and analyze");
-    }
-    release_prepared_stmt();
-    close_database();
-  }
-
-  bool insert_landmark(const Landmark& landmark);
-
+  LandmarkDatabase(const std::string& db_name, bool read_only);
+  void insert_landmark(const Landmark& landmark);
   std::vector<Landmark> get_landmarks_in_bounding_box(const double minLat,
```

Comment:
the interface becomes just the functions that we want to publicly expose. the only thing the outside caller can see that is protected is a pimpl which gives no clue about its implementation